src/components/network-map-tab.js

```diff
@@ -615,41 +614,53 @@ export const NetworkMapTab = ({
         // Hack to avoid reload Geo Data when switching display mode to TREE then back to MAP or HYBRID
         // TODO REMOVE LATER
         if (!reloadMapNeeded) return;
-        loadMapEquipments();
+        if (!isInitialized) {
+            loadMapEquipments();
+            loadAllGeoData();
+        } else {
+            updateMapEquipmentsAndGeoData();
+        }
         setInitialized(true);
         // Note: studyUuid and dispatch don't change
     }, [
         disabled,
         studyUuid,
         currentNode,
         loadMapEquipments,
+        updateMapEquipmentsAndGeoData,
+        loadAllGeoData,
         isInitialized,
         reloadMapNeeded,
         updatedSubstationsIds,
     ]);
 
-    // This useEffect ensures the geo data load is done after the map equipments load. Improvement: the sequentiality could be enforced by using a chain of promises for the updates of mapEquipments and geoData
-    // At initialization, loadMapEquipments and loadGeoData can be parallelized.
-    // Also, because of this useEffect instead of a chain of promises, we have to use a ref for currentNode in loadGeoData, loadAllGeoData and loadMissingGeoData Because currentNode would trigger loadGeoData and loadMapEquipments which retriggers loadGeoData.
+    // Reload geo data (if necessary) when we switch on full path
     useEffect(() => {
-        if (!mapEquipments) return;
-        loadGeoData();
-    }, [loadGeoData, mapEquipments]);
+        const prevLineFullPath = lineFullPathRef.current;
+        lineFullPathRef.current = lineFullPath;
+        if (isInitialized && lineFullPath && !prevLineFullPath) {
+            loadGeoData();
+        }
+    }, [isInitialized, lineFullPath, loadGeoData]);
```

Comment:
`isInitialized` change will proc this effect. Do we really need it ?
It is set to true just after a load or an update, then there is a case where `loadGeoData` will be called twice ?

Reply:
If we don't check with isIinitialized we fetch geo data 2 times at initialization if lineFullPath is true.
We could fix it by setting useRef(true) but using this boolean seems to be safer and clearer to me. Should I change it ?

Reply:
In this case, this is handled by explicitly checking for lineFullPath && !prevLineFullPath. The effect is triggered but it does nothing in other cases